cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperatorTest.java

```diff
@@ -246,6 +246,82 @@ private void krNewToProposalReady(VertxTestContext context, int pendingCalls, Cr
                 }));
     }
 
+    /**
+     * Tests the transition from 'NotReady' to 'ProposalReady' when
+     * the kafkaRebalance resource  spec is updated with "skip hard goals check" state
+     *
+     * 1. A new KafkaRebalance resource is created with some specified not hard goals; it is in the New state
+     * 2. The operator requests a rebalance proposal through the Cruise Control REST API
+     * 3. The operator receives a "missing hard goals" error instead of a proposal
+     * 4. The KafkaRebalance resource moves to the 'NotReady' state
+     * 5. The rebalance spec is updated with the 'skip hard goals check' field to "true"
+     * 6. The operator requests a rebalance proposal through the Cruise Control REST API
+     * 7. The rebalance proposal is ready on the first call
+     * 8. The KafkaRebalance resource moves to the 'ProposalReady' state
+     */
+    @Test
+    public void krNotReadyToProposalReadyOnSpecChange(VertxTestContext context) throws IOException, URISyntaxException {
```

Comment:
Maybe the name could start with `test` prefix to make it easier to find? `testKrNotReadyToProposalReadyOnSpecChange`?

Reply:
Done